backend/main.py

```diff
@@ -27,7 +27,11 @@
 
 
 async def scan_handler(*args):
-    await scan.scan(*args, sm)
+    try:
+        await scan.scan(sm, *args)
+    except Exception as err:
+        await sm.emit("scan:done_ko", str(err))
+        raise (err)
```

Comment:
I chose to re-raise the error here to keep with existing behaviour, though we can change that and just log it.

Reply:
I'm ok with this, better user experience